src/kOS/Module/kOSVesselModule.cs

```diff
@@ -223,16 +226,55 @@ private void HookEvents()
         /// </summary>
         private void UnHookEvents()
         {
-            if (RemoteTechHook.IsAvailable(parentVessel.id))
+            if (hasRemoteTech)
             {
-                RemoteTechHook.Instance.RemoveSanctionedPilot(parentVessel.id, UpdateAutopilot);
+                UnHookRemoteTechPilot();
             }
             else
             {
-                parentVessel.OnPreAutopilotUpdate -= UpdateAutopilot;
+                UnHookStockPilot();
             }
         }
 
+        private void CheckSwapEvents()
```

Comment:
(I edited the comment above several times, trying to change how I was describing it.  I'm done doing that now.  Sorry about all the edits while you might have been trying to respond to it.)